### PR TITLE
[WiP] Download Dashboards as PDF

### PR DESCRIPTION
**Pull request by ghost**
Working on being able to download a dashboard or slice as a PDF so visualizations can easily be shared/emailed/etc.

Still working on authentication (currently, the pdf created will show an image of the access denied and login screen), passing the correct dashboard to be downloaded, and checks on the system so the file can be saved to an appropriate folder.

Any feedback is greatly appreciated!

### CODE REVIEW

**Comment by SpyderRivera** [comment]
@abowden-spotx Can you add some detail around the way you are approaching this?

File: superset/assets/javascripts/dashboard/components/Header.jsx [hunk 1]
```diff
@@ -68,6 +69,19 @@ class Header extends React.PureComponent {
         {btnText}
       </Button>);
   }
+  renderDownloadButton() {
+    // TODO: review button style
+    const btnText = 'Download';
```
Comment (mistercrunch) [inline superset/assets/javascripts/dashboard/components/Header.jsx]: `t('My translated string')`

**Comment by mistercrunch** [comment]
The absolute easiest here would be to
1) hide the menu bar, this can be done with the `&standalone=true` URL param, but perhaps could be done in JS on the fly with some JQuery-hack
2) show the user how to `Print as PDF`, maybe just link to a website that explains that well with browser-specific detection and tutorial

**Comment by SpyderRivera** [comment]
#4721

**Comment by abowden-spotx** [comment]
Ultimately, the goal of this addition is so that we can integrate it with an emailing feature that will run independently on the server, which will allow us to send visualizations of the data to our ops teams.

So far, I have looked into:

- [pdfkit](https://pypi.org/project/pdfkit/) (Python wrapper for [wkhtmltopdf](https://wkhtmltopdf.org/))
- [jsPDF](https://github.com/MrRio/jsPDF)

**Comment by SpyderRivera** [comment]
@mistercrunch @williaster

**Pull request closed without merging**